### PR TITLE
Finish derived addresses, fix floating point issues, prepare 1.3.2114

### PR DESCRIPTION
**Pull request by MrStahlfelge**
* Derived addresses on receive screen
* Derived addresses on send screen
* Use BigDecimal for parsing ergo and token amounts from user inputs

### CODE REVIEW

File: app/src/main/java/org/ergoplatform/android/wallet/WalletDetailsFragment.kt [hunk 1]
```diff
@@ -143,15 +158,15 @@ class WalletDetailsFragment : Fragment(), AddressChooserCallback {
 
         // fill balances
         val addressState = address?.let { wallet.getStateForAddress(address) }
-        val ergoAmount = nanoErgsToErgs(
+        val ergoAmount = ErgoAmount(
             addressState?.balance
                 ?: wallet.getBalanceForAllAddresses()
         )
-        binding.walletBalance.amount = ergoAmount
+        binding.walletBalance.amount = ergoAmount.toDouble()
```
Comment (aslesarenko) [inline app/src/main/java/org/ergoplatform/android/wallet/WalletDetailsFragment.kt:165]: avoid double

File: app/src/main/java/org/ergoplatform/android/wallet/WalletDetailsFragment.kt [hunk 2]
```diff
@@ -143,15 +158,15 @@ class WalletDetailsFragment : Fragment(), AddressChooserCallback {
...
 
         val unconfirmed = addressState?.unconfirmedBalance
             ?: wallet.getUnconfirmedBalanceForAllAddresses()
-        binding.walletUnconfirmed.amount = nanoErgsToErgs(unconfirmed)
+        binding.walletUnconfirmed.amount = ErgoAmount(unconfirmed).toDouble()
```
Comment (aslesarenko) [inline app/src/main/java/org/ergoplatform/android/wallet/WalletDetailsFragment.kt:169]: avoid double

**Review by aslesarenko — APPROVED** [review]
LGTM, but please see comments

**Comment by MrStahlfelge** [comment]
Opened issue #41 to deal with the `ErgoAmount` remarks, will do the code formatting/naming issues with the next commit.

**Pull request merged**